src/scripts/runTestsCI.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    downloadAndUnzipVSCode,
+    resolveCliArgsFromVSCodeExecutablePath,
+    runTests
+} from "@vscode/test-electron";
+import * as cp from "child_process";
+import * as path from "path";
+
+const extensionDependencies = [
+    // Cursorless access to Tree sitter
+    "pokey.parse-tree",
+
+    // Register necessary language-IDs for tests
+    "mrob95.vscode-talonscript", // talon
+    "jrieken.vscode-tree-sitter-query" // scm
+];
+
+function installExtensionDependencies(vscodeExecutablePath: string) {
+    const [cli, ...args] = resolveCliArgsFromVSCodeExecutablePath(vscodeExecutablePath);
+
+    // Install extension dependencies
+    const extensionInstallArgs = [
+        ...args,
+        ...extensionDependencies.flatMap((dependency) => ["--install-extension", dependency])
+    ];
+
+    console.log("starting to install dependency extensions");
+    console.log(`cli: ${cli}`);
+    console.log(JSON.stringify(extensionInstallArgs, null, 2));
+
+    const { status, signal, error } = cp.spawnSync(cli, extensionInstallArgs, {
+        encoding: "utf-8",
+        stdio: "inherit"
+    });
+
+    console.log("status: ", status);
+    if (signal) {
+        console.log("signal: ", signal);
+    }
+    if (error) {
+        console.log("error: ", error);
+    }
+
+    console.log("finished installing dependency extensions\n");
+}
+
+export async function launchVscodeAndRunTests() {
+    try {
+        const outDirPath = path.join(__dirname, "../../out");
```

Comment:
This should point to root I believe

Reply:
setting `extensionDevelopmentPath` to root it didn't help with the  command not found errors

Reply:
Ok I guess either way you need to do that tho. It should point to the directory containing your package.json

Reply:
Maybe put a log statement in extension activation to be sure it's loading?

Reply:
I added a large debug print to the activate function and that is never run.

Reply:
I've found the problem. In package.json I had specified that my extension depends on the command server. I didn't install the command server  in the test environment so the extension just silently failed to activate. No logs, no nothing the extension just would not activate. . Even worst when I used `vscode.extensions.all` and looked after my extension it actually had the boolean `isActive` set to true.